rfcs/2020-04-04-2221-kubernetes-integration.md

```diff
@@ -0,0 +1,1625 @@
+          compression = "gzip"
+          region = "us-east-1"
+          key_prefix = "date=%F/"
+    CONFIG
+    ```
+
+2.  Deploy Vector!
+
+    Now that you have your custom `ConfigMap` ready it's time to deploy Vector.
+    Create a `Namespace` and apply your `ConfigMap` and our recommended
+    deployment configuration into it:
+
+    ```shell
+    kubectl create namespace vector
+    kubectl apply --namespace vector -f vector-configmap.yaml
+    kubectl apply -f https://packages.timber.io/vector/latest/kubernetes/vector-global.yaml
+    kubectl apply --namespace vector -f https://packages.timber.io/vector/latest/kubernetes/vector-namespaced.yaml
+    ```
+
+    - _See [outstanding questions 3, 4, 5, 6, and 7](#outstanding-questions)._
+
+    That's it!
+
+#### Deploy using Helm
+
+1.  Install [`helm`][helm_install].
+
+2.  Add our Helm Chart repo.
+
+    ```shell
+    helm repo add vector https://charts.vector.dev
+    helm repo update
+    ```
+
+3.  Configure Vector.
+
+    TODO: address this when we decide on the helm chart internals.
+
+4.  Deploy Vector!
+
+    ```shell
+    kubectl create namespace vector
+
+    # Helm v3
+    helm upgrade \
+      --install \
+      --namespace vector \
+      --values vector-values.yaml \
+      vector \
+      vector/vector
+
+    # Helm v2
+    helm upgrade
+      --install  \
+      --namespace vector \
+      --values vector-values.yaml \
+      --name vector \
+      vector/vector
+    ```
+
+#### Deploy using Kustomize
+
+1.  Install [`kustomize`][kustomize].
+
+1.  Prepare `kustomization.yaml`.
+
+    Use the same config as in [`kubectl` guide][anchor_tutorial_kubectl].
+
+    ```yaml
+    # kustomization.yaml
+    namespace: vector
+
+    resources:
+      - https://packages.timber.io/vector/latest/kubernetes/vector-global.yaml
+      - https://packages.timber.io/vector/latest/kubernetes/vector-namespaced.yaml
+      - vector-configmap.yaml
+    ```
+
+1.  Deploy Vector!
+
+    ```shell
+    kustomize build . | kubectl apply -f -
+    ```
+
+## Design considerations
+
+### Minimal supported Kubernetes version
+
+The minimal supported Kubernetes version is the earliest released version of
+Kubernetes that we intend to support at full capacity.
+
+We use minimal supported Kubernetes version (or MSKV for short), in the
+following ways:
+
+- to communicate to our users what versions of Kubernetes Vector will work on;
+- to run our Kubernetes test suite against Kubernetes clusters starting from
+  this version;
+- to track what Kubernetes API feature level we can use when developing Vector
+  code.
+
+We can change MSKV over time, but we have to notify our users accordingly.
+
+There has to be one "root" location where current MSKV for the whole Vector
+project is specified, and it should be a single source of truth for all the
+decisions that involve MSKV, as well as documentation. A good candidate for
+such location is a file at `.meta` dir of the Vector repo. `.meta/mskv` for
+instance.
```

Comment:
You mean `.meta/installation/platforms.toml`? I think it's a similar but different at it's core concern... I'm afraid we'll be mixing things that shouldn't mix, and it'd limit the flexibility of both file formats - so I'd go with a separate file.
Maybe something like `.meta/supported_versions.toml`

Reply:
Eh, I'm not so sure about that. The `.meta/installation/*.toml` is meant to be the authoritative list of everything we support. We also specific versions, or requirements, in the `.meta/{sources|transforms|sinks}/*.toml` files as well.

Reply:
Well, I think the only concern I have on where we manage that file is what automation we apply over it.
So far I wasn't able to pin the proper way of how we can derive, for example, our CI config, from that file. I think this will be more obvious when we start with test infra implementation.

Omitting that concern - I'm fine with keeping that data at `.meta/installation/platforms.toml` - but I'd like to hold the decision on that for now.